src/_pytest/python.py

```diff
@@ -645,6 +721,16 @@ def _get_xunit_setup_teardown(holder, attr_name, param_obj=None):
             return result
 
 
+def _call_with_optional_argument(func, arg):
+    arg_count = func.__code__.co_argcount
+    if inspect.ismethod(func):
+        arg_count -= 1
+    if arg_count:
```

Comment:
please add an assertion for arg_count being in `0,1`

Reply:
Good point, but actually `test_exact_teardown_issue1206` checks that we get a Python error message if we declare a setup/teardown function with the wrong arguments:

https://github.com/pytest-dev/pytest/blob/f3b6425324c99875e52c057cd4bb5c3959fd1e53/testing/test_runner.py#L309-L343

So I'm not adding the assertion. 👍